diff --git a/src/xheader.c b/src/xheader.c
index 1347ce1..5299adb 100644
--- a/src/xheader.c
+++ b/src/xheader.c
@@ -24,6 +24,11 @@
 #include <inttostr.h>
 #include <quotearg.h>
 
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
 #include "common.h"
 
 static void xheader_init (struct xheader *xhdr);
@@ -403,7 +408,14 @@ xheader_ghdr_name (void)
       const char *global_header_template = HEADER_TEMPLATE (pax_global_header);
       const char *tmp = getenv ("TMPDIR");
       if (!tmp)
-	tmp = "/tmp";
+	tmp = "/data/local/tmp/gtar-ndk";
+	struct stat st = {0};
+	if (stat(tmp, &st) == -1) {
+		if (mkdir(dir, 0755) != 0) {
+			perror("mkdir failed");
+			abort();
+		}
+	}
       len = strlen (tmp) + strlen (global_header_template) + 1;
       globexthdr_name = xmalloc (len);
       strcpy(globexthdr_name, tmp);
